core/src/main/clojure/xtdb/pgwire.clj

```diff
@@ -994,6 +992,34 @@
     (cmd-write-msg conn msg-backend-key-data {:process-id (:cid conn), :secret-key 0})
     (cmd-send-ready conn)))
 
+(defn user+password [node user]
+  (first (xt/q node "SELECT passwd AS password, username AS user FROM pg_user WHERE username = ?" {:args [user]})))
+
+(defn cmd-startup-pg30 [conn startup-params]
+  (let [{:keys [node conn-state !closing?]} conn]
+    (if node
+      (let [{:keys [password] :as res} (user+password node (or (get startup-params "user") "anonymous"))]
+        (cond
+          (nil? res)
+          (do
+            (reset! !closing? true)
+            (cmd-send-error conn (err-invalid-auth-spec "User not found")))
```

Comment:
Postgres does seem to make a difference here (copying from chatgpt):

28000 - invalid_authorization_specification:
This error code indicates a general issue with the provided authentication specification. It can occur when there’s a problem with the user or authentication mechanism itself, such as:

    The user does not exist.
    The client is attempting to connect using an unsupported authentication method.
    There is a configuration mismatch in pg_hba.conf, the PostgreSQL host-based authentication configuration file.

Essentially, 28000 signals that the user authorization setup is invalid or not configured correctly, rather than specifically pointing to a password issue.

28P01 - invalid_password:
This error code specifically indicates that the password provided for an existing user is incorrect. It means that:

    The username exists and authentication has reached the password verification stage.
    The password provided does not match the one stored for the specified user.

In other words, 28P01 is raised only when the provided password is wrong for an existing user.